manifests/server/config_entry.pp

```diff
@@ -10,7 +10,7 @@
 define postgresql::server::config_entry (
   Enum['present', 'absent']                               $ensure  = 'present',
   String[1]                                               $key     = $name,
-  Optional[Variant[String[1], Numeric, Array[String[1]]]] $value   = undef,
+  Optional[Variant[String, Numeric, Array[String[1]]]]    $value   = undef,
```

Comment:
Should the array also allow empty strings?

Reply:
I am not aware of a use-case where an array would contain empty string values. 

But I am not even sure if array values make sense for Postgres config entries, because it seems to use just (quoted) strings, integers, floats and booleans (https://www.postgresql.org/docs/current/config-setting.html#CONFIG-SETTING-NAMES-VALUES). And from quickly looking at the `postgresql_conf` provider I don't immediately see where/how it even handles array values? But in any case, I don't think we need to touch this part of the functionality here. Just support for single, empty string values would be nice :)

Reply:
Indeed, it seems the accepted data types of `$value` were changed to include `Array[String[1]]`, and the "parsed" provider was replaced by a "ruby" provider, but there doesn't seem to have been any code added to actually handle such array values. The parsed provider had something like ` h[:value].join(', ')` when `h[:value]` was an array, but I see no equivalent in the ruby provider.

But I suppose this discussion is best moved elsewhere, probably to its own bug report?

Reply:
> Indeed, it seems the accepted data types of `$value` were changed to include `Array[String[1]]`, and the "parsed" provider was replaced by a "ruby" provider, but there doesn't seem to have been any code added to actually handle such array values. The parsed provider had something like ` h[:value].join(', ')` when `h[:value]` was an array, but I see no equivalent in the ruby provider.

If the implementation is ` h[:value].join(', ')` then an empty array should result in an empty string and allowing empty strings within the array doesn't make sense. It'd result in `, ` which is probably never valid. In other words, consider my comments addressed.

> But I suppose this discussion is best moved elsewhere, probably to its own bug report?

Yes, agreed.